### PR TITLE
Polish the `MultiImage` docstring.

### PR DESCRIPTION
**Pull request by mkcor**
## Description

This minor PR builds on top of #6290 by @thvoigtmann (sorry, I couldn't review yesterday). This changeset is only intended to make the wording more standard and more direct. Notably,

* I removed "in the current implementation" because there is no reason to imply anything about a future implementation;
* since the object created by `ImageCollection` is named `ic`, I thought the one created by `MultiImage` might as well be named `mi` (rather than `img`).

## Checklist

- [Docstrings for all functions](https://github.com/numpy/numpy/blob/master/doc/example.py)
- Gallery example in `./doc/examples` (new features only)
- Benchmark in `./benchmarks`, if your changes aren't covered by an
  existing benchmark
- Unit tests
- Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)
- Descriptive commit messages (see below)

## For reviewers

- Check that the PR title is short, concise, and will make sense 1 year
  later.
- Check that new functions are imported in corresponding `__init__.py`.
- Check that new features, API changes, and deprecations are mentioned in
  `doc/release/release_dev.rst`.
- There is a bot to help automate backporting a PR to an older branch. For
  example, to backport to v0.19.x after merging, add the following in a PR
  comment: `@meeseeksdev backport to v0.19.x`
- To run benchmarks on a PR, add the `run-benchmark` label. To rerun, the label
  can be removed and then added again. The benchmark output can be checked in
  the "Actions" tab.

### CODE REVIEW

File: skimage/io/collection.py [hunk 1]
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
...
     For a TIFF image containing N frames of size WxH, `MultiImage` stores
-    all frames of that image as a single entry of shape `(N, W, H)` in the
-    list. `ImageCollection` instead creates N entries of shape `(W, H)`.
+    all frames of that image as a single element of shape `(N, W, H)` in the
+    list. `ImageCollection` instead creates N elements of shape `(W, H)`.
 
-    For an animated GIF image, `MultiImage` in the current implementation
-    will only read one frame, while `ImageCollection` by default will read
-    all of them.
+    For an animated GIF image, `MultiImage` reads only the first frame, while
```
Comment (mkcor) [inline skimage/io/collection.py:427]: That's what I noticed with `no_time_for_that_tiny.gif` -- hoping it's not any random one that `MultiImage` loads...

Comment (thvoigtmann) [reply skimage/io/collection.py:427]: @mkcor for the GIFs that I have tried, `MultiImage` indeed seems to load the first frame. At the moment I am not even sure `ImageCollection` does the right thing for animated GIF. Could it be that it detects all the frames but always returns the first one? I will try to look into this in the context of [#6291](https://github.com/scikit-image/scikit-image/pull/6291) once I have time.

**Review by grlee77 — APPROVED** [review]
The doc fixes here look good.

It seems there is still some question about the expected behavior for animated GIFs, though, so let's wait a bit before merge to see if that can be resolved. Alternatively, if a separate issue is opened for that we can just merge this now.

File: skimage/io/collection.py [hunk 2]
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
 
     Notes
     -----
-    The object that is returned can be used as a list of image-data objects,
-    where each entry in the list represents one image. In this regard it is
-    very similar to `ImageCollection`, but the two differ in the treatment
-    of multi-frame images.
+    The object that is returned is a list of image-data arrays. In this
```
Comment (alexdesiqueira) [inline skimage/io/collection.py:419]: What if:
```suggestion
    `MultiImage` returns a list of image-data arrays. In this
```
Then the other sentence (`In this regard, it is...`) would make more sense, I guess.

File: skimage/io/collection.py [hunk 3]
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
...
+    regard, it is very similar to `ImageCollection`, but the two differ in
+    their treatment of multi-frame images.
 
     For a TIFF image containing N frames of size WxH, `MultiImage` stores
-    all frames of that image as a single entry of shape `(N, W, H)` in the
-    list. `ImageCollection` instead creates N entries of shape `(W, H)`.
+    all frames of that image as a single element of shape `(N, W, H)` in the
+    list. `ImageCollection` instead creates N elements of shape `(W, H)`.
```
Comment (alexdesiqueira) [inline skimage/io/collection.py:425]:
```suggestion
    list. `ImageCollection`, in its turn, creates N elements of shape `(W, H)`.
```

File: skimage/io/collection.py [hunk 4]
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
...
 
-    For an animated GIF image, `MultiImage` in the current implementation
-    will only read one frame, while `ImageCollection` by default will read
-    all of them.
+    For an animated GIF image, `MultiImage` reads only the first frame, while
+    `ImageCollection` reads all of them by default.
```
Comment (alexdesiqueira) [inline skimage/io/collection.py:428]:
```suggestion
    `ImageCollection` reads all frames by default.
```

File: skimage/io/collection.py [hunk 5]
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
...
 
     Examples
     --------
     >>> from skimage import data_dir
 
     >>> multipage_tiff = data_dir + '/multipage.tif'
-    >>> img = MultiImage(multipage_tiff)
-    >>> len(img) # img contains one file
+    >>> mi = MultiImage(multipage_tiff)
```
Comment (alexdesiqueira) [inline skimage/io/collection.py:435]: I'm personally not very fond of `mi` here. How about something more descriptive? :slightly_smiling_face: 
```suggestion
    >>> multi_img = MultiImage(multipage_tiff)
```

File: skimage/io/collection.py [hunk 6]
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
...
+    >>> len(mi)  # mi contains one element
```
Comment (alexdesiqueira) [inline skimage/io/collection.py:436]: If you agree with the previous comment,
```suggestion
    >>> len(multi_img)  # multi_img contains one element
```

File: skimage/io/collection.py [hunk 7]
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
...
     1
-    >>> img[0].shape # this image contains two frames of size (15, 10)
+    >>> mi[0].shape  # this element is a two-frame image of shape (2, 15, 10)
```
Comment (alexdesiqueira) [inline skimage/io/collection.py:438]:
```suggestion
    >>> multi_img[0].shape  # this element is a two-frame image of shape (2, 15, 10)
```

File: skimage/io/collection.py [hunk 8]
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
...
     (2, 15, 10)
 
     >>> ic = ImageCollection(multipage_tiff)
```
Comment (alexdesiqueira) [inline skimage/io/collection.py:441]: I'd say the same here. How about something more descriptive?
```suggestion
    >>> image_col = ImageCollection(multipage_tiff)
```
I know lots of our documentation have `ic` around, but I'll get there eventually :slightly_smiling_face:

File: skimage/io/collection.py [hunk 9]
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
...
-    >>> len(ic) # ic contains two images
+    >>> len(ic)  # ic contains two elements
```
Comment (alexdesiqueira) [inline skimage/io/collection.py:442]: If...
```suggestion
    >>> len(image_col)  # image_col contains two elements
```

File: skimage/io/collection.py [hunk 10]
```diff
@@ -416,39 +416,36 @@ class MultiImage(ImageCollection):
...
     2
     >>> for frame in ic:
```
Comment (alexdesiqueira) [inline skimage/io/collection.py:444]:
```suggestion
    >>> for frame in image_col:
```

**Review by alexdesiqueira — CHANGES_REQUESTED** [review]
Thank you @mkcor @thvoigtmann! I've suggested a few things that improve readability in my mind; please feel free to agree with or discuss them.

**Comment by alexdesiqueira** [comment]
@grlee77 could you check if the changes I requested make sense, and add them if they do? :slightly_smiling_face: I think this one could be nicely merged after that. Thank you!

**Comment by thvoigtmann** [comment]
@alexdesiqueira @mkcor Thanks for the updates, looking good to me.

@grlee77 Just to clarify regarding the animated GIFs, I think the purpose here is just to match the docstring with what actually happens. (Here: only the first frame of the GIF is loaded.) If that is "expected behavior" can be debated :-)

I'm happy to have you all commenting on [PR #6291](https://github.com/scikit-image/scikit-image/pull/6291) where I think I've managed to resolve the issue. Please see the "Updated Description" there. This is still marked as "draft" because I am a bit short on coding time these days and there is still a failing test that I don't understand.

File: skimage/io/collection.py (hunk 2 above)
Comment (mkcor) [reply skimage/io/collection.py:419]: You're absolutely right, thank you!

File: skimage/io/collection.py (hunk 3 above)
Comment (mkcor) [reply skimage/io/collection.py:425]: I don't think so... Your wording would suggest that `ImageCollection` is called at some point.

File: skimage/io/collection.py (hunk 4 above)
Comment (mkcor) [reply skimage/io/collection.py:428]: Improves clarity indeed, thanks.

File: skimage/io/collection.py (hunk 5 above)
Comment (mkcor) [reply skimage/io/collection.py:435]: Ok!

File: skimage/io/collection.py (hunk 8 above)
Comment (mkcor) [reply skimage/io/collection.py:441]: I agree with your suggestions, I think they make the documentation more readable :relaxed:

**Comment by mkcor** [comment]
Hi @alexdesiqueira,

Thank you so much for the detailed review! I have finally addressed your suggestions. When you squash and merge, feel free to collapse the last commit messages (me not having updated my local setup yet :roll_eyes:).

**Review by alexdesiqueira — APPROVED** [review]
I'm happy. Thank you @mkcor!

**Comment by thvoigtmann** [comment]
Oh wow, a burst of activity today. Thank you :)

**Pull request merged**